doc/use_cases.md

```diff
@@ -0,0 +1,65 @@
+# Use case scenarios for posteriordb
+
+posteriordb has different uses. Below is list of example use cases and what features are important in each use case.
+
+## Unit testing
```

Comment:
Based on the description the better name would be "System testing"

Reply:
So I tried to collect all testing under one title. Maybe call it "Unit, system and regression testing"? Or just "Testing"?

Reply:
I don't think posteriordb would be very useful in unit testing.
I think it's important to mention both system and regression testing as the first one requires the true reference result and the latter just the previous version result.

Reply:
I fully agree, unit testing is less of a use case, indeed. So do you think ”Testing” would be a sufficient header with regression and system testing as sub headers?

Reply:
Maybe "Software testing" with regression and system testing as sub headers?

Reply:
Sounds good! Ill fix.